src/proto/h2/client.rs

```diff
@@ -110,6 +111,7 @@ where
                             if let Some(len) = body.content_length() {
                                 headers::set_content_length_if_missing(req.headers_mut(), len);
                             }
+                            let content_length = content_length_parse_all(req.headers());
```

Comment:
This looks like it sets `content_length` of the **Response**'s body by parsing the length from the **Request**. (Curious the tests didn't catch it...)

Reply:
It looks like it does pick it up but I suspect that because the panic is not in the main test thread it doesn't cause the test to fail. 

https://travis-ci.org/hyperium/hyper/jobs/390534326#L825
https://travis-ci.org/hyperium/hyper/jobs/390534326#L825

I'll update the tests to fix this and remove:

```rust
if let Some(len) = body.content_length() {
  headers::set_content_length_if_missing(req.headers_mut(), len);
}
```

from the implementation.